test/fail_compilation/fail4544.d

```diff
@@ -0,0 +1,23 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/fail4544.d(15): Error: unterminated character constant
+fail_compilation/fail4544.d(16): Error: `0x` isn't a valid integer literal, use `0x0` instead
+fail_compilation/fail4544.d(16): Error: no identifier for declarator `int`
+fail_compilation/fail4544.d(17): Error: unterminated character constant
+fail_compilation/fail4544.d(18): Error: unterminated character constant
+---
+*/
+
+int foo(char n, int m)
+{
+    int k = 5;
+    char c = 'asd';
```

Comment:
Instead of:
```
fail_compilation/fail4544.d(15): Error: unterminated character constant
```
A better error message here would be:
```
fail_compilation/fail4544.d(15): Error: character constant has multiple characters
```
Is there a way to do this?

Reply:
This would take a separate check because there's also the case `char c = 'aaa; ` in which the "unterminated character constant" is a good error message.